### PR TITLE
Fix build issue on 32bit system

### PR DESCRIPTION
**Pull request by PradeepKiruvale**
## Proposed changes

Fix the build issue on 32bit system. Use usize instead of u64.

## Types of changes

What types of changes does your code introduce to `thin-edge.io`?
_Put an `x` in the boxes that apply_

- [x ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Improvement (general improvements like code refactoring that doesn't explicitly fix a bug or add any new functionality)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Paste Link to the issue
<br/>

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [ ] I have read the [CONTRIBUTING](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTING.md) doc
- [ ] I have signed the [CLA](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTOR-LICENSE-AGREEMENT.md) (in all commits with git commit -s)
- [ ] I ran `cargo fmt` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I used `cargo clippy` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: sm/download/src/download.rs [hunk 1]
```diff
@@ -123,15 +123,15 @@ fn create_file_and_try_pre_allocate_space(
...
+            if file_len >= usable_disk_space as usize {
                 return Err(DownloadError::InsufficientSpace);
             }
             // Reserve diskspace
             let _ = fallocate(
                 file.as_raw_fd(),
                 FallocateFlags::empty(),
                 0,
-                file_len as i64,
+                file_len as libc::off_t,
```
Comment (didier-wenzek) [inline sm/download/src/download.rs:134]: Why not an `isize` ?

File: sm/download/Cargo.toml [hunk 2]
```diff
@@ -9,6 +9,7 @@ description = "download_manager"
 [dependencies]
 backoff = { version = "0.3", features = ["tokio"] }
 json_sm = { path = "../json_sm" }
+libc = "0.2.107"
```
Comment (didier-wenzek) [inline sm/download/Cargo.toml:12]: I'm surprised to see that dependency just for a single type def.

* Why is the proper type not managed by the `nix` crate?

File: sm/download/src/download.rs (hunk 1 above)
Comment (PradeepKiruvale) [reply sm/download/src/download.rs]: Tried. it does not work.

File: sm/download/Cargo.toml (hunk 2 above)
Comment (PradeepKiruvale) [reply sm/download/Cargo.toml]: Fixed, replaced with nix::libc::off_t

**Review by didier-wenzek — APPROVED** [review]
Approved

**Pull request merged**